.github/workflows/CI.yml

```diff
@@ -29,104 +35,194 @@ jobs:
           toolchain: nightly
           override: true
           components: clippy
+
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y libgeos-dev
       - run: cargo clippy -- -D warnings
       - run: cd sys && cargo clippy -- -D warnings
 
   check_static:
     name: Check static feature
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     env:
       LD_LIBRARY_PATH: /usr/local/lib
+
     steps:
       - uses: actions/checkout@v2
         with:
-          submodules: 'true'
+          submodules: "true"
+
       - name: Install toolchain
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: stable
           override: true
-      - name: Install dependencies
-        run: |
-          sudo apt update
-          sudo apt install valgrind
-          sudo apt remove libgeos-dev -y
-      - name: Install geos
-        # First we install the 3.8 version of libgeos and then checkout the 3.8.1 version
-        run: |
-          git clone https://github.com/libgeos/geos
-          cd geos
-          git checkout 24650422b3982f17cc493fe92a70228f2ad624b4
-          ./autogen.sh
-          ./configure
-          make
-          sudo make install
-          sudo ldconfig
+
       - name: Build static geos crate
         run: cargo build --features static
 
   check:
-    name: Check ${{ matrix.toolchain }} / ${{ matrix.triple.target }}
-    runs-on: ubuntu-latest
+    name: Check ${{ matrix.toolchain }} / ${{ matrix.geos }} - ${{ matrix.triple.target }}
+    runs-on: ubuntu-20.04
     env:
       LD_LIBRARY_PATH: /usr/local/lib
+      GEOS_LIB_DIR: "/usr/lib/x86_64-linux-gnu"
+      GEOS_VERSION: ${{ matrix.geos }}
     strategy:
       fail-fast: false
       matrix:
         toolchain:
           - stable
           - nightly
+        geos:
+          - "3.6.5"
+          - "3.7.5"
+          - "3.8.1"  # TEMP: since 3.8.1 worked previously
+          - "3.8.3"
+          - "3.9.3"
+          - "3.10.3"
+          - "3.11.0beta1"
```

Comment:
I don't think it's a good idea to test not yet stable versions.

Reply:
In other projects based on GEOS (e.g., [Shapely](https://github.com/shapely/shapely/blob/main/.github/workflows/tests.yml#L17) - python bindings), we test against GEOS `main` branch as part of the CI suite, but we allow those to fail without failing the suite.  This has helped us catch incoming changes that break our tests (most often due to GEOS bug fixes or changes in coordinate order).

However, that's a bit separate than how I have it here: `3.11.0beta1` is intended to be a placeholder that gets updated to 3.11.0 when released (should be soon?) or gets removed.  Sorry, I should have flagged this part as needing to be revisited prior to merge.

Reply:
Please don't add support for not yet stable versions. It's much simpler to just add its support once its released rather that updating code if it has breaking change between the currently supported version and the one that got actually stabilized.